src/torrent.cpp

```diff
@@ -8244,16 +8244,16 @@ namespace libtorrent
 			m_seeding_time += lost_seconds;
 		}
 		m_became_seed = clamped_subtract_u16(m_became_seed, seconds);
-
-		if (m_finished_time < seconds && is_finished())
+		if (m_finished_time < seconds && is_finished() && !is_paused())
```

Comment:
I don't know what is correct because the var names does confuse me.

m_finished_time seems to be used to tell how long a torrent was in finish state and m_became_finished is the time at which we go into finish state.

That means we need to add lost seconds only if it is not paused during time switch.

I am not sure if this is correct, but m_seeding_time use is_seed() which means not paused as well and m_active_time does have `!is_paused()` too. After the change all 3 times are handled equally. If the new semantics is wrong, then m_seeding_time and m_active_time is wrong too.

Maybe it's more easy to understand what we get over the interface, and without skiping pause like at the other 2 times we got a wrong value at st.finished_duration.